third_party/move/move-vm/types/src/values/values_impl.rs

```diff
@@ -1591,31 +1606,36 @@ impl IntegerValue {
         Ok(match self {
             U8(x) => {
                 if n_bits >= 8 {
-                    return Err(PartialVMError::new(StatusCode::ARITHMETIC_ERROR));
+                    return Err(PartialVMError::new(StatusCode::ARITHMETIC_ERROR)
```

Comment:
I am wondering whether we can use a static string constant to represent the error string such as "Shift Left overflow"?

Reply:
I don't think you can do that? We only have `&'static str` in rust, according to https://stackoverflow.com/questions/32956050/how-to-repeat-a-static-string-at-compile-time

Reply:
Sorry for the confusion. I mean something like this: https://github.com/aptos-labs/aptos-core/blob/e0a42ece1909baa8298b24b8e4a12010825051a3/crates/aptos-inspection-service/src/server/mod.rs#L42C7-L42C7

Reply:
Yea my point is even if you define err msgs as `const` you still need to invoke `.to_string()` which at this point I'm not sure how that differs from having a string literal like what we have for now.

Reply:
A possibility to consider is a private method that takes a string slice and returns the appropriate `Err` object: this may reduce some code duplication.

I also learnt that Rust already interns duplicate string literals, so there should be no binary size penalty for using duplicate string literals.

Reply:
I suspect that Teng means that putting that literal into a const means less work if you decide to change it.  In this case, you could separate the error generation from the shifting, at the cost of an extra match:
```
 match (match self {
            U8(x) => {
                if n_bits >= 8 {
                    Err(1)
                } else {
                    Ok(IntegerValue::U8(x << n_bits))
                },
           ...
}) {
    OK(value) => Ok(value),
    Error(_) => Err(PartialVMError::new(StatusCode::ARITHMETIC_ERROR)
                        .with_message("Shift Left overflow".to_string())),
}
```